test/unit/models/taskRequest.test.js

```diff
@@ -0,0 +1,167 @@
+const chai = require("chai");
+const { expect } = chai;
+
+const cleanDb = require("../../utils/cleanDb");
+const userData = require("../../fixtures/user/user")();
+const addUser = require("../../utils/addUser");
+const taskModel = require("../../../models/tasks");
+const taskData = require("../../fixtures/tasks/tasks")();
+const taskRequestModel = require("../../../models/taskRequests");
+const userModel = require("../../../models/users");
+const { TASK_STATUS } = require("../../../constants/tasks");
+const { TASK_REQUEST_STATUS } = require("../../../constants/taskRequests");
+
+describe("Task Request", function () {
+  afterEach(async function () {
+    await cleanDb();
+  });
+
+  describe("addOrUpdate", function () {
+    let user0, user1, task;
+
+    beforeEach(async function () {
+      user0 = await addUser(userData[9]);
+      user1 = await addUser(userData[10]);
+      task = await taskModel.updateTask(taskData[4]);
+    });
+
+    it("should add the task if no taskRequest is present for taskId", async function () {
+      const result = await taskRequestModel.addOrUpdate(task.taskId, user0);
+
+      expect(result).to.be.an("object");
+      expect(result).to.have.property("isCreate");
+      expect(result.isCreate).to.be.eq(true);
+      expect(result).to.have.property("taskRequest");
+      expect(result.taskRequest).to.be.an("object");
+      expect(result.taskRequest).to.have.property("requestors");
+      expect(result.taskRequest.requestors).to.have.length(1);
+      expect(result.taskRequest.requestors[0]).to.be.eq(user0);
+      expect(result.taskRequest).to.have.property("status");
+      expect(result.taskRequest.status).to.be.eq(TASK_REQUEST_STATUS.WAITING);
+      expect(result).to.have.property("id");
```

Comment:
few expect statements seem redundant, you could directly make do with 
`expect(result.taskRequest.requestors[0]).to.be.eq(user0);`

Reply:
Please do it ad other places as well

Reply:
use something like expect(object).to.have.all.keys([key array])

Reply:
done